src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefConditionalOperatorTests.cs

```diff
@@ -0,0 +1,799 @@
+  IL_001a:  brtrue.s   IL_0028
+  IL_001c:  ldsflda    ""int C.val2""
+  IL_0021:  call       ""ref int C.M1(ref int)""
+  IL_0026:  br.s       IL_002d
+  IL_0028:  ldsflda    ""int C.val1""
+  IL_002d:  dup
+  IL_002e:  ldind.i4
+  IL_002f:  ldc.i4.s   22
+  IL_0031:  add
+  IL_0032:  stind.i4
+  IL_0033:  ldsfld     ""int C.val2""
+  IL_0038:  call       ""void System.Console.Write(int)""
+  IL_003d:  ret
+}
+");
+        }
+
+        [Fact]
+        public void TestRefConditionalOperatorIncrementUsed()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        bool b = false;
+        System.Console.Write((b? ref val1: ref val2) ++);
+        System.Console.Write((b? ref val1: ref val2) += 22);
+
+        System.Console.Write(val2);
+    }
+
+    static int val1 = 33;
+    static int val2 = 44;
+}";
+            var comp = CompileAndVerify(source, expectedOutput: "446767", verify: false);
+            comp.VerifyDiagnostics();
+
+            comp.VerifyIL("C.Main", @"
+{
+  // Code size       68 (0x44)
+  .maxstack  4
+  .locals init (int V_0)
+  IL_0000:  ldc.i4.0
+  IL_0001:  dup
+  IL_0002:  brtrue.s   IL_000b
+  IL_0004:  ldsflda    ""int C.val2""
+  IL_0009:  br.s       IL_0010
+  IL_000b:  ldsflda    ""int C.val1""
+  IL_0010:  dup
+  IL_0011:  ldind.i4
+  IL_0012:  stloc.0
+  IL_0013:  ldloc.0
+  IL_0014:  ldc.i4.1
+  IL_0015:  add
+  IL_0016:  stind.i4
+  IL_0017:  ldloc.0
+  IL_0018:  call       ""void System.Console.Write(int)""
+  IL_001d:  brtrue.s   IL_0026
+  IL_001f:  ldsflda    ""int C.val2""
+  IL_0024:  br.s       IL_002b
+  IL_0026:  ldsflda    ""int C.val1""
+  IL_002b:  dup
+  IL_002c:  ldind.i4
+  IL_002d:  ldc.i4.s   22
+  IL_002f:  add
+  IL_0030:  dup
+  IL_0031:  stloc.0
+  IL_0032:  stind.i4
+  IL_0033:  ldloc.0
+  IL_0034:  call       ""void System.Console.Write(int)""
+  IL_0039:  ldsfld     ""int C.val2""
+  IL_003e:  call       ""void System.Console.Write(int)""
+  IL_0043:  ret
+}
+");
+        }
+
+        [Fact]
+        public void TestRefConditionalOperatorInRefAssignment()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        bool b = true;
+        int x = 1;
+        int y = 2;
+
+        ref var local = ref b ? ref x : ref y;
+
+        System.Console.WriteLine(local);
+    }
+}";
+            var comp = CompileAndVerify(source, expectedOutput: "1");
+            comp.VerifyDiagnostics();
+            comp.VerifyIL("C.Main", @"
+{
+  // Code size       20 (0x14)
+  .maxstack  2
+  .locals init (int V_0, //x
+                int V_1) //y
+  IL_0000:  ldc.i4.1
+  IL_0001:  ldc.i4.1
+  IL_0002:  stloc.0
+  IL_0003:  ldc.i4.2
+  IL_0004:  stloc.1
+  IL_0005:  brtrue.s   IL_000b
+  IL_0007:  ldloca.s   V_1
+  IL_0009:  br.s       IL_000d
+  IL_000b:  ldloca.s   V_0
+  IL_000d:  ldind.i4
+  IL_000e:  call       ""void System.Console.WriteLine(int)""
+  IL_0013:  ret
+}
+");
+        }
+
+        // elvis target reg and generic
```

Comment:
Idea for couple more tests. Will add.

---
In reply to: [105271954](https://github.com/dotnet/roslyn/pull/17436#discussion_r105271954) [](ancestors = 105271954)